tls/s2n_x509_validator.c

```diff
@@ -319,7 +319,7 @@ s2n_cert_validation_code s2n_x509_validator_validate_cert_chain(struct s2n_x509_
 
         struct s2n_blob asn1cert = {0};
         asn1cert.size = certificate_size;
-        asn1cert.data = s2n_stuffer_raw_read(&cert_chain_in_stuffer, certificate_size);
+        notnull_check(asn1cert.data = s2n_stuffer_raw_read(&cert_chain_in_stuffer, certificate_size));
         if (asn1cert.data == NULL) {
```

Comment:
Shouldn't this be an `S2N_ERROR_IF`?

Reply:
Ahh, in fact, this function returns an enum, not the standard -1 plus setting the error code.  So it shouldn't be.  But it is confiusing to have a function that doesn't follow the standard s2n usage pattern.

Reply:
yeah this signature seems to be returning `s2n_cert_validation_code`. maybe @JonathanHenson would know about this.

Reply:
It looks like grep_simple_mistakes.sh is relying on `\s` to check for spaces [here](https://github.com/awslabs/s2n/blame/master/.travis/grep_simple_mistakes.sh#L57-L59). That won't work on OSX, because \s is a GNU extension. 

Escaping the single space, or allowing for any blank, would make the check pass on OSX and Linux, while keeping the intention of the regex the same. I'm assuming it's ok to not use the `notnull_check` function since the script verifies an explicit NULL check is present.

```sh
diff --git a/.travis/grep_simple_mistakes.sh b/.travis/grep_simple_mistakes.sh
index f65f4a5a..1003dc2f 100755
--- a/.travis/grep_simple_mistakes.sh
+++ b/.travis/grep_simple_mistakes.sh
@@ -54,7 +54,7 @@ for file in $S2N_FILES_ASSERT_NOTNULL_CHECK; do
     # $line_one definitely contains an assignment from s2n_stuffer_raw_read(),
     # because that's what we grepped for. So verify that either $line_one or
     # $line_two contains a null check.
-    manual_null_check_regex=".*if.*==\sNULL"
+    manual_null_check_regex=".*if.*==\ NULL"
     if [[ $line_one == *"notnull_check("* ]] || [[ $line_one =~ $manual_null_check_regex ]] ||\
     [[ $line_two == *"notnull_check("* ]] || [[ $line_two =~ $manual_null_check_regex ]]; then
       # Found a notnull_check
```

Above is the fix I had in mind. I'm just adding it as a patch because it seems more straightforward than a new PR linking to this PR.